RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift

```diff
@@ -86,9 +105,243 @@ extension NonFungibleTokenDetails {
 				}
 				.foregroundColor(.app.gray1)
 				.task { @MainActor in
-					await viewStore.send(.view(.task)).finish()
+					await viewStore.send(.task).finish()
+				}
+			}
+		}
+
+		@ViewBuilder
+		private func arbitraryDataFieldView(_ field: ViewState.TokenDetails.ArbitraryDataField, viewStore: ViewStoreOf<NonFungibleTokenDetails>) -> some SwiftUI.View {
+			switch field.kind {
+			case let .primitive(value):
+				ViewThatFits(in: .horizontal) {
+					KeyValueView(key: field.name, value: value)
+					VStack(alignment: .leading, spacing: .small3) {
+						Text(field.name)
+							.textStyle(.body1Regular)
+							.foregroundColor(.app.gray2)
+						ExpandableTextView(
+							fullText: value
+						)
+						.textStyle(.body1HighImportance)
+						.foregroundColor(.app.gray1)
+					}
+					.flushedLeft
+				}
+
+			case .complex:
+				KeyValueView(key: field.name, value: L10n.AssetDetails.NFTDetails.complexData)
+
+			case let .url(url):
+				VStack(alignment: .leading, spacing: .small3) {
+					Text(field.name)
+						.textStyle(.body1Regular)
+						.foregroundColor(.app.gray2)
+					Button(url.absoluteString) {
+						viewStore.send(.openURLTapped(url))
+					}
+					.buttonStyle(.url)
 				}
+				.flushedLeft
+
+			case let .address(address):
+				KeyValueView(key: field.name) {
+					AddressView(.address(address))
+				}
+
+			case let .decimal(value):
+				KeyValueView(key: field.name, value: value.formatted())
+
+			case let .enum(variant):
+				KeyValueView(key: field.name, value: variant)
+
+			case let .id(id):
+				KeyValueView(key: field.name, value: try! id.toString())
```

Comment:
can we avoid force `try!` ?

Reply:
`if let idString = try? id.toString() {
    KeyValueView(...)
} else {
   // what goes here?
}

Reply:
sure, though it is assured to not fail. Can fallback to an empty view.